realm/src/androidTest/java/io/realm/RealmQueryTest.java

```diff
@@ -459,4 +473,45 @@ public void testGeorgian() {
             assertEquals(sorted[i], stringOnlies2.get(i).getChars());
         }
     }
+
+    public void testQueryNullStringNotNullableField() {
+        try {
+            RealmResults<NullTypes> nullTypeses = testRealm.where(NullTypes.class).equalTo(NullTypes.FIELD_STRING_NOT_NULL,
+                    (String)null).findAll();
```

Comment:
Yes, as the compiler cannot deduce if I'm calling the `String` or `Date` version of the method.